gossip/src/contact_info.rs

```diff
@@ -335,7 +334,7 @@ impl ContactInfo {
             }
         }
         if let Some(entry) = self.cache.get_mut(usize::from(key)) {
-            *entry = socket;
+            *entry = sanitize_socket(&socket).map(|()| socket);
```

Comment:
can we just do this:
```suggestion
            *entry = Ok(socket);
```
since we call:
```
sanitize_socket(&socket)?;
```
on line 309? the socket has already been sanitized.

Reply:
right. done. thanks